### PR TITLE
sql-statements: add doc of lock tables and unlock tables

### PR DESCRIPTION
**Pull request by crazycs520**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Close https://github.com/pingcap/docs/issues/8632

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@crazycs520 Could you invite a technical review?

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+`LOCK TABLES` 可以为当前客户端会话获取表锁。你可以获取普通表的表锁，但你必须拥有锁定对象的 `LOCK TABLES` 和 `SELECT ` 权限。
+
+`UNLOCK TABLES` 显式释放当前会话持有的所有表锁。`LOCK TABLES` 在获取新锁之前会隐式释放当前会话持有的所有表锁。
+
+表锁可以防止其他会话的读取或写入。持有 `WRITE` 锁的会话可以执行表级操作，例如 `DROP TABLE` 或 `TRUNCATE TABLE`。
+
+> **警告：**
+>
+> `LOCK TABLES` 和 `UNLOCK TABLES` 目前为实验特性，不建议在生产环境中使用。
+> 开启表锁功能需要在所有 TiDB 的配置文件中设置 [enable-table-lock](/tidb-configuration-file.md#enable-table-lock-从-v400-版本开始引入) 为 `true`。
```
Comment (shichun-0415) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:19]:
```suggestion
> 开启表锁功能需要在所有 TiDB 的配置文件中设置 [`enable-table-lock`](/tidb-configuration-file.md#enable-table-lock-从-v400-版本开始引入) 为 `true`。
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+不能在单个 `LOCK TABLES` 语句中多次获取同一个表的锁。
+
+```sql
+> LOCK TABLE t WRITE, t READ;
+(1066, "Not unique table/alias: 't'")
+```
+
+## 释放表锁
+
+释放会话持有的表锁时，将同时释放该会话所有持有的表锁。会话可以显式释放表锁，也可以在以下情况下隐式释放表锁。
```
Comment (shichun-0415) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:76]:
```suggestion
释放会话持有的表锁时，将同时释放该会话所有持有的表锁。会话可以显式释放表锁，也可以在以下情况下隐式释放表锁：
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+- 会话可以使用 `UNLOCK TABLES` 语句显式释放其持有的所有表锁。
+- 当会话使用 `LOCK TABLES` 语句来获取表锁，而且同时已经持有其他表锁时，则在获取新表锁之前，将隐式释放其已经持有的所有表锁。
+
+当客户端会话的连接终止（无论是正常还是异常），TiDB 都会隐式释放会话持有的所有表锁。如果客户端重新连接，锁将不再有效。因此，客户端可能希望禁用自动重新连接。自动重新连接生效后，如果发生重新连接，但任何表锁或当前事务都将丢失，而且不会通知客户端。禁用自动重新连接后，如果连接断开，则客户端会话发出的下一条语句将会收到报错信息。客户端可以检测到错误并采取适当的操作，例如重新获取锁或重做事务。
```
Comment (shichun-0415) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:81]:
```suggestion
当客户端会话的连接终止（无论是正常还是异常），TiDB 都会隐式释放会话持有的所有表锁。如果客户端重新连接，锁将不再有效。因此，客户端可能希望禁用自动重新连接。自动重新连接生效后，如果发生重新连接，任何表锁或当前事务都将丢失，而且不会通知客户端。禁用自动重新连接后，如果连接断开，则客户端会话发出的下一条语句将会收到报错信息。客户端可以检测到错误并采取适当的操作，例如重新获取锁或重做事务。
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+## 表锁的使用限制和条件
+
+你可以安全地使用 `KILL` 语句终止已经持有表锁的会话。
+
+你不能获取 `INFORMATION_SCHEMA`，`PERFORMANCE_SCHEMA`，`METRICS_SCHEMA` 和 `mysql` 数据库中的表的表锁。
```
Comment (shichun-0415) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:87]:
```suggestion
你不能获取 `INFORMATION_SCHEMA`、`PERFORMANCE_SCHEMA`、`METRICS_SCHEMA` 和 `mysql` 数据库中表的表锁。
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+## 和 MySQL 的兼容性
+
+### 获取表锁
+
+- 在 TiDB 中，如果会话 A 已经持有了一个表的表锁，另一个会话 B 对该表写入时会报错；但 MySQL 会阻塞会话 B 对该表的写入，直到会话 A 释放该表锁。
+其他会话对该表的锁请求会被阻塞直到当前会话释放 `WRITE` 锁。
+- 在 TiDB 中，如果 `LOCK TABLES` 语句想要获取的表锁被其他会话持有且必须等待锁释放时，则 `LOCK TABLES` 语句会执行报错；但 MySQL 会阻塞 `LOCK TABLES` 语句的执行直到成功获取想要的表锁。
+- 在 TiDB 中，使用 `LOCK TABLES` 语句获取表锁的作用域是整个集群生效的；但 MySQL 中表锁的作用域是单个 MySQL 服务器，与NDB群集不兼容。
```
Comment (shichun-0415) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:96]:
```suggestion
- 在 TiDB 中，如果 `LOCK TABLES` 语句想要获取的表锁被其他会话持有且必须等待锁释放时，`LOCK TABLES` 语句会执行报错；但 MySQL 会阻塞 `LOCK TABLES` 语句的执行，直到成功获取想要的表锁。
- 在 TiDB 中，使用 `LOCK TABLES` 语句获取表锁的作用域是整个集群；但 MySQL 中表锁的作用域是单个 MySQL 服务器，与 NDB 群集不兼容。 
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+### 获取表锁
+
+- 在 TiDB 中，如果会话 A 已经持有了一个表的表锁，另一个会话 B 对该表写入时会报错；但 MySQL 会阻塞会话 B 对该表的写入，直到会话 A 释放该表锁。
+其他会话对该表的锁请求会被阻塞直到当前会话释放 `WRITE` 锁。
+- 在 TiDB 中，如果 `LOCK TABLES` 语句想要获取的表锁被其他会话持有且必须等待锁释放时，则 `LOCK TABLES` 语句会执行报错；但 MySQL 会阻塞 `LOCK TABLES` 语句的执行直到成功获取想要的表锁。
+- 在 TiDB 中，使用 `LOCK TABLES` 语句获取表锁的作用域是整个集群生效的；但 MySQL 中表锁的作用域是单个 MySQL 服务器，与NDB群集不兼容。 
+
+### 释放表锁
+
+在 TiDB 的会话中显示开启一个事务时（例如用 `BEGIN` 语句），TiDB 不会隐式释放当前会话已经持有的表锁；但 MySQL 会隐式释放当前会话已经持有的表锁。
```
Comment (shichun-0415) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:100]:
```suggestion
在 TiDB 的会话中显示开启一个事务时（例如使用 `BEGIN` 语句），TiDB 不会隐式释放当前会话已经持有的表锁；但 MySQL 会隐式释放当前会话已经持有的表锁。
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+ERROR 8020 (HY000): Table 't1' was locked in WRITE by server: f4799bcb-cad7-4285-8a6d-23d3555173f1_session: 2199023255959
+```
+
+上面的错误信息说明在 TiDB `f4799bcb-cad7-4285-8a6d-23d3555173f1` 中会话 ID 为 `2199023255959` 的会话已经持有表 `t1` 的 `WRITE` 锁，所以当前会话无法获取表 `t1` 的 `READ` 锁。
+
+不能在单个 `LOCK TABLES` 语句中多次获取同一个表的锁。
+
+```sql
+> LOCK TABLE t WRITE, t READ;
+(1066, "Not unique table/alias: 't'")
```
Comment (bb7133) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:71]:
```suggestion
ERROR 1066 (42000): Not unique table/alias: 't'
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md (hunk 3 above)
Comment (bb7133) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:81]: > 因此，客户端可能希望禁用自动重新连接。

"客户端" 并不是真正的行为主体，这里是不是改为 "通常不建议在客户端使用自动重新连接：因为在开启自动连接时，如果发生重新连接..."

Comment (crazycs520) [reply sql-statements/sql-statement-lock-tables-and-unlock-tables.md]: addressed.

File: TOC.md [hunk 8]
```diff
@@ -802,6 +802,7 @@
       - [`SPLIT REGION`](/sql-statements/sql-statement-split-region.md)
       - [`START TRANSACTION`](/sql-statements/sql-statement-start-transaction.md)
       - [`TABLE`](/sql-statements/sql-statement-table.md)
+      - [`LOCK TABLES` 和 `UNLOCK TABLES`](sql-statements/sql-statement-lock-tables-and-unlock-tables.md)
```
Comment (shichun-0415) [inline TOC.md:805]:
```suggestion
      - [`LOCK TABLES` 和 `UNLOCK TABLES`](/sql-statements/sql-statement-lock-tables-and-unlock-tables.md)
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+## 释放表锁
+
+释放会话持有的表锁时，将同时释放该会话所有持有的表锁。会话可以显式释放表锁，也可以在以下情况下隐式释放表锁：
+
+- 会话可以使用 `UNLOCK TABLES` 语句显式释放其持有的所有表锁。
+- 当会话使用 `LOCK TABLES` 语句来获取表锁，而且同时已经持有其他表锁时，则在获取新表锁之前，将隐式释放其已经持有的所有表锁。
+
+当客户端会话的连接终止（无论是正常还是异常），TiDB 都会隐式释放会话持有的所有表锁。如果客户端重新连接，锁将不再有效。因此，通常不建议在客户端使用自动重新连接，因为在开启自动重新连接时，如果发生重新连接，任何表锁或当前事务都将丢失，而且不会通知客户端。禁用自动重新连接后，如果连接断开，则客户端会话发出的下一条语句将会收到报错信息。客户端可以检测到错误并采取适当的操作，例如重新获取锁或重做事务。
+
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:85]:
```suggestion

你可以安全地使用 `KILL` 语句终止已经持有表锁的会话。

```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+## 表锁的使用限制和条件
+
+你可以安全地使用 `KILL` 语句终止已经持有表锁的会话。
+
+你不能获取 `INFORMATION_SCHEMA`、`PERFORMANCE_SCHEMA`、`METRICS_SCHEMA` 和 `mysql` 数据库中表的表锁。
+
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:91]:
```suggestion
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+`LOCK TABLES` 可以为当前客户端会话获取表锁。你可以获取普通表的表锁，但你必须拥有锁定对象的 `LOCK TABLES` 和 `SELECT` 权限。
+
+`UNLOCK TABLES` 显式释放当前会话持有的所有表锁。`LOCK TABLES` 在获取新锁之前会隐式释放当前会话持有的所有表锁。
+
+表锁可以防止其他会话的读取或写入。持有 `WRITE` 锁的会话可以执行表级操作，例如 `DROP TABLE` 或 `TRUNCATE TABLE`。
+
+> **警告：**
+>
+> 开启表锁功能需要在所有 TiDB 实例的配置文件中设置 [`enable-table-lock`](/tidb-configuration-file.md#enable-table-lock-从-v400-版本开始引入) 为 `true`。
+
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:23]:
```suggestion
## 使用限制

不支持获取以下数据库中表的表锁：

- `INFORMATION_SCHEMA`
- `PERFORMANCE_SCHEMA`
- `METRICS_SCHEMA` 
- `mysql`

```

File: tidb-configuration-file.md [hunk 12]
```diff
@@ -207,7 +207,7 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 
 + 控制是否开启表级锁特性。
 + 默认值：false
-+ 表级锁用于协调多个 session 之间对同一张表的并发访问。目前已支持的锁种类包括 `READ`、`WRITE` 和 `WRITE LOCAL`。当该配置项为 `false` 时，执行 `LOCK TABLE` 和 `UNLOCK TABLE` 语句不会生效，并且会报 "LOCK/UNLOCK TABLES is not supported" 的警告。
++ 表级锁用于协调多个 session 之间对同一张表的并发访问。目前已支持的锁种类包括 `READ`、`WRITE` 和 `WRITE LOCAL`。当该配置项为 `false` 时，执行 `LOCK TABLE` 和 `UNLOCK TABLE` 语句不会生效，并且会报 "LOCK/UNLOCK TABLES is not supported" 的警告。更多信息，请参考 [`LOCK TABLES` 和 `UNLOCK TABLES`](/sql-statements/sql-statement-lock-tables-and-unlock-tables.md)。
```
Comment (hfxsd) [inline tidb-configuration-file.md:210]:
```suggestion
+ 表级锁用于协调多个 session 之间对同一张表的并发访问。目前已支持的锁种类包括 `READ`、`WRITE` 和 `WRITE LOCAL`。当该配置项为 `false` 时，执行 `LOCK TABLES` 和 `UNLOCK TABLES` 语句不会生效，并且会报 "LOCK/UNLOCK TABLES is not supported" 的警告。更多信息，请参考 [`LOCK TABLES` 和 `UNLOCK TABLES`](/sql-statements/sql-statement-lock-tables-and-unlock-tables.md)。
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md (hunk 11 above)
Comment (crazycs520) [reply sql-statements/sql-statement-lock-tables-and-unlock-tables.md:23]: 感觉不用特地把这个使用限制放在这个前面来，他不算是非常重要的前提。 MySQL 文档里面也没有把这种限制说明放在最前面来

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md (hunk 9 above)
Comment (crazycs520) [reply sql-statements/sql-statement-lock-tables-and-unlock-tables.md:85]: MySQL 那边是把这句话放到 `## 表锁的使用限制和条件` 章节里面的

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md (hunk 11 above)
Comment (hfxsd) [reply sql-statements/sql-statement-lock-tables-and-unlock-tables.md:23]: OK 你们自己定夺就行

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+不能在单个 `LOCK TABLES` 语句中多次获取同一个表的锁。
+
+```sql
+> LOCK TABLE t WRITE, t READ;
+ERROR 1066 (42000): Not unique table/alias: 't'
+```
+
+## 释放表锁
+
+释放会话持有的表锁时，将同时释放该会话所有持有的表锁。会话可以显式释放表锁，也可以在以下情况下隐式释放表锁：
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:79]:
```suggestion
释放会话持有的表锁时，将同时释放该会话所有持有的表锁。会话可以显式或隐式释放表锁：
```

File: sql-statements/sql-statement-lock-tables-and-unlock-tables.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+> lock table t1 read;
+ERROR 8020 (HY000): Table 't1' was locked in WRITE by server: f4799bcb-cad7-4285-8a6d-23d3555173f1_session: 2199023255959
+```
+
+以上错误信息表明，在 TiDB `f4799bcb-cad7-4285-8a6d-23d3555173f1` 中，ID 为 `2199023255959` 的会话已经持有表 `t1` 的 `WRITE` 锁，所以，当前会话无法获取表 `t1` 的 `READ` 锁。
+
+不能在单个 `LOCK TABLES` 语句中多次获取同一个表的锁。
+
+```sql
+> LOCK TABLE t WRITE, t READ;
```
Comment (hfxsd) [inline sql-statements/sql-statement-lock-tables-and-unlock-tables.md:73]: 我们文档中的示例都用的是 LOCK TABLE 而不是 LOCK TABLES，和 MySQL 文档中的不同 https://dev.mysql.com/doc/refman/8.0/en/lock-tables.html 。所以我们文档中所有的 LOCK TABLES 和 UNLOCK TABLES 是否应该全部改为单数 LOCK TABLE 和 UNLOCK TABLE？

Comment (crazycs520) [reply sql-statements/sql-statement-lock-tables-and-unlock-tables.md:73]: 我改下这个示例吧，其实 SQL 语句里面用 `LOCK TABLE` 或者 `LOCK TABLES` 都是一样可以的

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**